gdscript/sidebar.gd

```diff
@@ -660,49 +656,95 @@ func update_ui(update_diff: bool = false) -> void:
 			change_author = "Anonymous"
 
-		var change_timestamp = human_readable_timestamp(change.timestamp)
+		var item = history_tree.create_item(root)
+		history_item_count += 1
 
-		var prefix = ""
+		# set columns
+		var column_index = 0
+		var base_column_count = 2
 
+		# if we're a dev, we need another column for the commit hash
 		if DEV_MODE:
-			prefix = change.hash.substr(0, 8) + " - "
+			history_tree.columns = base_column_count + 1
+			item.set_text(column_index, change.hash.substr(0, 8))
+			item.set_tooltip_text(column_index, change.hash)
+			item.set_selectable(0, false)
+			history_tree.set_column_expand(0, true)
+			history_tree.set_column_expand_ratio(0, 0)
+			history_tree.set_column_custom_minimum_width(0, 80)
+			column_index += 1
+		else:
+			history_tree.columns = base_column_count
+
+		item.set_metadata(0, change.hash)
+		# enabled flag
+		item.set_metadata(1, true)
+		history_tree.set_column_expand(column_index, true)
+		history_tree.set_column_expand_ratio(column_index, 2)
+		item.set_selectable(column_index, true)
 
-		var idx = -1
-		# Disable the last two items from being selected
-		var is_disabled = checked_out_branch and checked_out_branch.is_main and i < 2
-		# print(change)
 		if "merge_metadata" in change:
 			var merged_branch = GodotProject.get_branch_by_id(change.merge_metadata.merged_branch_id)
 			var merged_branch_name = str(change.merge_metadata.merged_branch_id)
 			if merged_branch:
 				merged_branch_name = merged_branch.name
-			idx = history_list.add_item(prefix + "↪️ " + change_author + " merged \"" + merged_branch_name + "\" branch - " + change_timestamp, item_context_menu_icon)
-			history_list.set_item_metadata(idx, change.hash)
+			item.set_text(column_index, "↪️ " + change_author + " merged \"" + merged_branch_name + "\" branch")
+			item.add_button(column_index, load("res://addons/patchwork/icons/branch-icon-history.svg"), 0, false, "Checkout branch " + merged_branch_name)
 
 		elif "reverted_to" in change:
 			var reverted_to: PackedStringArray = change.reverted_to
 			for j in range(reverted_to.size()):
 				# just truncate the hash to 7 characters
 				reverted_to[j] = reverted_to[j].substr(0, 7)
-			idx = history_list.add_item(prefix + "↩️ " + change_author + " reverted to " + ", ".join(reverted_to) + " - " + change_timestamp, item_context_menu_icon)
-			history_list.set_item_metadata(idx, change.hash)
-		else:
-			idx = history_list.add_item(prefix + change_author + " made some changes - " + change_timestamp + "", item_context_menu_icon)
-			history_list.set_item_metadata(idx, change.hash)
+			item.set_text(column_index, "↩️ " + change_author + " reverted to " + ", ".join(reverted_to))
 
-		history_list.set_item_disabled(idx, is_disabled)
+		else:
+			item.set_text(column_index, change_author + " made some changes")
 
 		if unsynced_changes.has(change.hash):
-			history_list.set_item_custom_fg_color(idx, Color(0.5, 0.5, 0.5))
+			item.set_custom_color(column_index, Color(0.6, 0.6, 0.6))
```

Comment:
Does this make the custom color apply to all the cells in the tree item, or just this cell?